faceted_search/src/index.js

```diff
@@ -1,65 +1,159 @@
+		{label: "Material", field: "hasMaterialCategory", type: "list-facet", facetSort:"count"},
+		{label: "Specimen", field: "hasSpecimenCategory", type: "list-facet", facetSort:"count"},
+		{label: "Registrant", field: "registrant", type: "list-facet", facetSort:"count"},
 ];
 
 // The sortable fields you want
 const sortFields = [
-	{label: "Name", field: "koppelnaam_s"},
-	{label: "Date of birth", field: "birthDate_i"},
-	{label: "Date of death", field: "deathDate_i"}
+		{label: "Identifier", field: "id"},
+		{label: "Registrant", field: "registrant"},
 ];
 
 
 // Construct the solr client api class
-const solrClient = 	new SolrClient({
-	url: config.solr_url,
-	searchFields: fields,
-	sortFields: sortFields,
+const solrClient = new SolrClient({
+		url: config.solr_url,
+		searchFields: fields,
+		sortFields: sortFields,
+		rows: 20,
+		pageStrategy: "paginate",
+		start: 0,
 
-	// Delegate change callback to redux dispatcher
-	onChange: (state) => store.dispatch({type: "SET_SOLR_STATE", state: state})
+		// Delegate change callback to redux dispatcher
+		onChange: (state) => {store.dispatch({type: "SET_SOLR_STATE", state: state})}
 });
 
+function APP() {
+	// https://reactrouterdotcom.fly.dev/docs/en/v6/api#usesearchparams
+	// Used for modifying the query string
+    let [searchParams, setSearchParams] = useSearchParams();
+	const Ref = useRef(null);
+
+	// A note about when this gets called -- https://reactjs.org/docs/hooks-reference.html#useeffect
+	// This is called asynchronously after a render is complete.  The rule is that render itself must be
+	// stateless, so state mutating operations need to be contained here.  In our case the sequence is
+	// (1) User clicks something in the UI (like enables or disables a search facet )
+	// (2) UI refreshes
+	// …some time passes
+	// (3) This gets called, and we write back the current query string to the browser's location using setSearchParams
+    useEffect(() => {
+		// For now, encode only the selected search facets and start page in the searchParams
+        let searchFields = encode(JSON.stringify(store.getState()['query']['searchFields']));
+        let start = encode(JSON.stringify(store.getState()['query']['start']/store.getState()['query']['rows']));
+		let sortFields = encode(JSON.stringify(store.getState()['query']['sortFields']));
+		let searchParamsDict = {"searchFields": searchFields, "start": start, "sortFields": sortFields};
+		// These are the arbitrary way to get text search value
+		let allTextFieldsValue = store.getState()['query']['searchFields'][0];
+		let identiFierValue = store.getState()['query']['searchFields'][1];
+		
+		// Update the query parameters with the latest values selected in the UI
+        setSearchParams(searchParamsDict);
+		highlight("All text fields", allTextFieldsValue ,Ref.current);
+		convertToLink('Identifier', "https://n2t.net/", Ref.current);
+		highlight("Identifier", identiFierValue ,Ref.current);
+		
```

Comment:
Use our own solr library extension to highlight and convert it to the link. But still no idea about indents